app/src/main/java/com/jerboa/ui/components/home/HomeActivity.kt

```diff
@@ -84,282 +73,149 @@ fun HomeActivity(
     val scrollBehavior = TopAppBarDefaults.enterAlwaysScrollBehavior(rememberTopAppBarState())
     val ctx = LocalContext.current
     val account = getCurrentAccount(accountViewModel)
+    val bottomNavController = rememberNavController()
+
+    // The bottomNavController remembers which route to show. Therefore, the selectedTab also
+    // has to be persisted across navigations to ensure that the tab selected is also the route
+    // being shown.
+    var selectedTab by rememberSaveable { mutableStateOf(selectTabArg) }
+    LaunchedEffect(account) {
+        // Required for the inbox deeplink.
+        if (selectedTab.needsLogin() && account == null) {
+            selectedTab = Route.HomeArgs.TAB_DEFAULT
+        }
+    }
+
+    val onSelectTab = { tab: HomeTab ->
+        if (tab.needsLogin() && account == null) {
+            feedNavController.toLogin.navigate()
+        } else {
```

Comment:
This is a change in behavior though, it will navigate to the login instead of displaying a toast like before. Not sure if this is wanted.

Reply:
One of the frequent complaints by new users is that it is difficult to find the option to log in. Redirecting users to log in when they attempt to use a function that requires being logged in seems like a good solution to that problem that many have requested.

Reply:
Well one of the other suggestions where to add a login button in the more options dialog and/or show login snackbar on initial app use.

It still doesn't solve the root problem as new users still won't know where to switch/log out.

And now is inconsistent with the other buttons that require login(upvotes , ...) that still display display this toast.

Either way, the current dropdown should be made more clear. Maybe we could ve a "Click here to login" text next to anonymous. That only shows on initial startup and diseapears once you click on the anonymous drop down.

Reply:
For the bottom nav bar, the nav controller was already available so went ahead with navigating to the login page. For the upvotes, downvotes and reply, the nav controller (or callbacks) have to be passed deeper down the tree. There's a TODO to show the login page along with the toast, so I assume this will be revisited.

https://github.com/dessalines/jerboa/blob/b8e0dfb113e0e4ef40dd539ebbaed5ee3aedf05f/app/src/main/java/com/jerboa/Utils.kt#L116-L119

Agree with the log out thing.